src/pyatsimagebuilder/utils.py

```diff
@@ -242,70 +258,85 @@ def to_image_path(path, context, workspace_dir):
     return path
 
 
-def discover_jobs(jobfiles, context, install_dir, workspace_dir):
+def discover_jobs(jobfiles, 
+                  context_path, 
+                  install_dir, 
+                  workspace_dir, 
+                  relative_path=True):
     """ Discover job files based on regex
 
     Arguments:
         jobfiles (dict): Dict of jobfiles config
-        context (Context): container build context
+        context (Path): context.path
         install_dir (Path): installation directory
         workspace_dir (str): image workspace directory
+        relative_path (bool): To enable path trimming or not; Default to True
     """
     logger.info('Discovering Jobfiles')
 
     jobfiles.setdefault('match', DEFAULT_JOB_REGEXES)
 
     # 1. find all the job files in context by regex pattern
-    discovered_jobs = context.search_regex(jobfiles['match'], [
-        install_dir,
-    ])
+    discovered_jobs = search_regex(jobfiles['match'], 
+                                   context_path, 
+                                   [install_dir])
 
     # 2. find all job files by glob
     for pattern in jobfiles.get('glob', []):
-        discovered_jobs.extend(context.search_glob(pattern))
+        discovered_jobs.extend(context_path.rglob(pattern))
 
     # 3. find all job files by specificy paths
     for path in jobfiles.get('paths', []):
-        path = context.path / path
+        path = context_path / path
 
         if path.exists() and path.is_file():
             discovered_jobs.append(path)
 
     # 4. discover all files that are pyats job by marker
     discovered_jobs.extend(
-        filter(is_pyats_job, context.search_glob('*.py')))
+        filter(is_pyats_job, context_path.rglob('*.py')))
 
     # sort and remove duplicates
     discovered_jobs = sorted(set(discovered_jobs))
 
-    # compute path from context to image path
-    rel_job_paths = [to_image_path(i, context, workspace_dir) for i in discovered_jobs]
+    if relative_path:
+        # compute path from context to image path
+        job_paths = [to_image_path(i, 
+                                context_path, 
+                                workspace_dir) for i in discovered_jobs]
+    else:
+        job_paths = [str(i) for i in discovered_jobs]
 
-    # write the files into a file as json
-    context.write_file(install_dir / 'jobfiles.txt', json.dumps({'jobs': rel_job_paths}))
+    with open(install_dir / 'jobfiles.txt', 'w') as f:
+        json.dump({'jobs': job_paths}, f)
 
-    logger.info('Number of discovered job files: %s' % len(rel_job_paths))
+    logger.info('Number of discovered job files: %s' % len(job_paths))
     logger.info('List of job files written to: %s' % jobfiles)
 
 
-def discover_manifests(context, install_dir):
+def discover_manifests(path, install_dir, relative_path=True):
```

Comment:
`relative_path` should be `None` or an actual path here. Otherwise, I would have to search the entirety of `/tmp/something/` to use that as a relative path/context, whereas I might want `/tmp/something/` to be the relative path/context but only actually search within `/tmp/something/scripts/`.